v0.5/tools/submission/submission-checker.py

```diff
@@ -0,0 +1,238 @@
+"""
+A checker for mlperf inference submissions
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import collections
+import json
+import logging
+import os
+import re
+import sys
+import time
+
+# pylint: disable=missing-docstring
+
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("main")
+
+VALID_MODELS = ["ssd-small", "ssd-large", "mobilenet", "resnet", "gnmt"]
+REQUIRED_DIRECTORIES = ["code", "results", "measurements", "systems"]
+REQUIRED_PERF_FILES = ["mlperf_log_accuracy.json", "mlperf_log_summary.txt", "mlperf_log_detail.txt"]
+REQUIRED_ACC_FILES = REQUIRED_PERF_FILES + ["accuracy.txt"]
+REQUIRED_MEASURE_FILES = ["mlperf.conf", "README.md"]
+
+
+def get_args():
+    """Parse commandline."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input", required=True, help="submission directory")
+    args = parser.parse_args()
+    return args
+
+
+def path_to_dict(path):
+    d, f = {}, {}
+    for dirName, subdirList, fileList in os.walk(path):
+        d[dirName] = sorted(subdirList)
+        f[dirName] = sorted(fileList)
+    return d, f
+
+
+def check_accuracy(model, dir):
+    is_valid = False
+    # look for: accuracy=... or mAP=...
+    with open(os.path.join(dir, "accuracy.txt"), "r") as f:
+        for line in f:
+            m = re.match("^accuracy=(\d+\.\d+).*", line)
+            if m:
+                is_valid = True
+                break
+            m = re.match("^mAP=(\d+\.\d+).*", line)
+            if m:
+                is_valid = True
+                break
+                # TODO: we might as well check the total count
```

Comment:
We need to match GNMT accuracy result as well: `r"BLEU: ([0-9\.]+)"`

Reply:
yes